Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -3659,6 +3660,18 @@ internal static void UpdateFileData(Dictionary<AbsolutePath, FileOutputData> map
                                 continue;
                             }
 
+                            if (!OperatingSystemHelper.IsUnixOS)
+                            {
+                                var pathstring = access.ToString(pathTable);
+                                if (Directory.Exists(pathstring))

```

Comment:
One problem with `Directory.Exists()` is that a pip can produce a directory symlink that points to a non-existent file.  In that case, `Directory.Exists()` will (presumably) return false.  That's why I'd prefer to use `GetFileAttributes` instead and then check that both `ReparsePoint` and `Directory` flags are set